pinot-common/src/main/java/com/linkedin/pinot/common/data/Schema.java

```diff
@@ -27,64 +31,55 @@
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 import org.apache.helix.ZNRecord;
-import org.codehaus.jackson.JsonNode;
-import org.codehaus.jackson.JsonParseException;
 import org.codehaus.jackson.annotate.JsonIgnore;
 import org.codehaus.jackson.annotate.JsonIgnoreProperties;
-import org.codehaus.jackson.map.JsonMappingException;
 import org.codehaus.jackson.map.ObjectMapper;
-import org.json.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import com.linkedin.pinot.common.data.FieldSpec.DataType;
-import com.linkedin.pinot.common.data.FieldSpec.FieldType;
-import static com.linkedin.pinot.common.utils.EqualityUtils.hashCodeOf;
-import static com.linkedin.pinot.common.utils.EqualityUtils.isEqual;
-import static com.linkedin.pinot.common.utils.EqualityUtils.isNullOrNotSameClass;
-import static com.linkedin.pinot.common.utils.EqualityUtils.isSameReference;
 
 /**
- * Schema is defined for each column. To describe the details information of columns.
- * Three types of information are provided.
- * 1. the data type of this column: int, long, double...
- * 2. if this column is a single value column or a multi-value column.
- * 3. the real world business logic: dimensions, metrics and timeStamps.
- * Different indexing and query strategies are used for different data schema types.
+ * The <code>Schema</code> class is defined for each table to describe the details of the table's fields (columns).
+ * <p>Three field types are supported: DIMENSION, METRIC, TIME.
+ * ({@link com.linkedin.pinot.common.data.DimensionFieldSpec}, {@link com.linkedin.pinot.common.data.MetricFieldSpec},
+ * {@link com.linkedin.pinot.common.data.TimeFieldSpec})
+ * <p>For each field, a {@link com.linkedin.pinot.common.data.FieldSpec} is defined to provide the details of the field.
+ * <p>There could be multiple DIMENSION or METRIC fields, but at most 1 TIME field.
+ * <p>In pinot, we store data using 5 <code>DataType</code>s: INT, LONG, FLOAT, DOUBLE, STRING. All other
+ * <code>DataType</code>s will be converted to one of them.
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
```

Comment:
Remember last time we got exceptions while loading the schema? This ignore properties was added at that time to make it backward compatible. (Maybe there are some extra fields in old version schema)